lib/src/supabase_stream_builder.dart

```diff
@@ -139,16 +253,19 @@ class SupabaseStreamBuilder extends Stream<SupabaseStreamEvent> {
 
   Future<void> _getStreamData() async {
     _streamData = [];
+    String? realtimeFilter;
+    if (_streamFilter != null) {
```

Comment:
Good work!
I usually think it's fine to use a variable like this, but `streamFilter` is being used a lot that I think null checking would be useful. 


```suggestion
  Future<void> _getStreamData() async {
      _StreamPostgrestFilter? currentStreamFilter = _streamFilter;
    _streamData = [];
    String? realtimeFilter;
    if (currentStreamFilter != null) {
```

And then replace `_streamFilter` with `currentStreamFilter` and you get to remove all the `!` everywhere.